internal/log/log_test.go

```diff
@@ -2,365 +2,56 @@ package log
 
 import (
 	"bytes"
-	"errors"
-	"fmt"
-	"regexp"
-	"strings"
+	"context"
+	"log/slog"
 	"testing"
 
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
-	"go.uber.org/zap/zaptest/observer"
+	"github.com/stretchr/testify/require"
 )
 
-const ExpectedFilename = "log_test.go"
-
-func TestSetLevel(t *testing.T) {
-
-	type tcase struct {
-		isDefault bool
-		lvl       Level
-		tst       func() bool
-	}
-	fn := func(tc tcase) (string, func(*testing.T)) {
-
-		name := "Default"
-		if !tc.isDefault {
-			name = tc.lvl.String()
-		}
-
-		return name, func(t *testing.T) {
-
-			if !tc.isDefault {
-				l := level
-				SetLogLevel(tc.lvl)
-				// Restore the level back
-				defer SetLogLevel(l)
+func TestStackTraceHandler_TableDriven(t *testing.T) {
+	testCases := []struct {
+		name        string
+		level       slog.Level
+		message     string
+		expectStack bool
+	}{
+		{name: "Debug", level: slog.LevelDebug, message: "debug", expectStack: false},
+		{name: "Info", level: slog.LevelInfo, message: "info", expectStack: false},
+		{name: "Warn", level: slog.LevelWarn, message: "warn", expectStack: false},
+		{name: "Error", level: slog.LevelError, message: "error", expectStack: true},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var buf bytes.Buffer
+			baseHandler := slog.NewTextHandler(&buf, &slog.HandlerOptions{
+				Level:     slog.LevelDebug,
+				AddSource: true,
+			})
+			handler := &Handler{handler: baseHandler}
+			logger := slog.New(handler)
+
+			// Log based on level.
+			switch tc.level {
+			case slog.LevelDebug:
+				logger.Debug(tc.message)
+			case slog.LevelInfo:
+				logger.Info(tc.message)
+			case slog.LevelWarn:
+				logger.Warn(tc.message)
+			case slog.LevelError:
+				logger.Error(tc.message)
+			default:
+				logger.Log(context.Background(), tc.level, tc.message)
 			}
 
-			if tc.tst() {
-				t.Errorf("%v , expected level to be set correctly, got %v", name, level)
+			output := buf.String()
+			if tc.expectStack {
+				require.Contains(t, output, "stack", "Error log should include a stack trace")
+			} else {
+				require.NotContains(t, output, "stack", "Non-error log should not include a stack trace")
```

Comment:
Why not use `strings.Contains` or a regular expression:

```
stackRegx := regex.MustCompile(`\w+stack\w+`)
stackRegx.Match(output) 

```

Reply:
Force of habit really. There’s lots of good test helpers in testify and it’s simple and super readable. Larger codebases, with different people working on them tend to be very heterogeneous when it comes to testing. But I’m alright if you both want to remain on std libraries. 👍

Reply:
changed and dropped testify. still think testify makes it more readable. :)

Reply:
@iwpnd I'm not opposed to testify. I was more asking if you thought the dependency was worth it. ;-)